pypeit/pypeit.py

```diff
@@ -572,33 +573,34 @@ def extract_one(self, frames, det, bg_frames=[], std_outfile=None):
 
         # Do one iteration of object finding, and sky subtract to get initial sky model
         self.sobjs_obj, self.nobj, skymask_init = \
-            self.redux.find_objects(self.sciimg, self.sciivar, std=self.std_redux, ir_redux=self.ir_redux,
-                                    std_trace=std_trace,maskslits=self.maskslits,
+            self.redux.find_objects(std=self.std_redux, ir_redux=self.ir_redux,
+                                    std_trace=std_trace, maskslits=self.maskslits,
                                     show=self.show & (not self.std_redux),
                                     manual_extract_dict=manual_extract_dict)
 
         # Global sky subtraction, first pass. Uses skymask from object finding step above
         self.initial_sky = \
-            self.redux.global_skysub(self.sciimg, self.sciivar, self.caliBrate.tilts_dict['tilts'], skymask=skymask_init,
+            self.redux.global_skysub(self.caliBrate.tilts_dict['tilts'], skymask=skymask_init,
                                     std=self.std_redux, maskslits=self.maskslits, show=self.show)
 
         if not self.std_redux:
             # Object finding, second pass on frame *with* sky subtraction. Show here if requested
             self.sobjs_obj, self.nobj, self.skymask = \
-                self.redux.find_objects(self.sciimg - self.initial_sky, self.sciivar, std=self.std_redux, ir_redux=self.ir_redux,
+                self.redux.find_objects(std=self.std_redux, ir_redux=self.ir_redux,
```

Comment:
By suppressing the arguments to these codes you are obfuscating what they are doing and making the code harder to read. These arguments were kept for a reason. You are effectively passing them in again as the "skysub_img" optional argument so you are not saving any space, and you are creating less readable code. I am in favor of these functions having the key arguments be explicit for code readability and request that you change back to the original codes.

Reply:
I disagree.  By requiring these get passed in 
over and again it makes it seem like they 
are changing and we are not taking advantage
of the existence of this object.